pkg/machinery/client/resolver/roundrobin.go

```diff
@@ -5,30 +5,28 @@
 package resolver
 
 import (
-	"fmt"
 	"math/rand"
+	"net"
+	"strconv"
 	"strings"
 
-	"github.com/talos-systems/net"
 	"google.golang.org/grpc/resolver"
+
+	"github.com/talos-systems/talos/pkg/machinery/generic/slices"
 )
 
-// RegisterRoundRobinResolver registers round-robin gRPC resolver for specified port and returns scheme to use in grpc.Dial.
-func RegisterRoundRobinResolver(port int) (scheme string) {
-	scheme = fmt.Sprintf(roundRobinResolverScheme, port)
+// RoundRobinResolverScheme is a scheme to use in grpc.Dial for the round-robin gRPC resolver.
+// This resolver requires that all endpoints have a port appended.
+// To ensure this, use EnsureEndpointsHavePorts before constructing a connection string.
+const RoundRobinResolverScheme = "roundrobin"
```

Comment:
I believe etcd or something else might use a similar name, probably it might be safer to use `talosroundrobin` here.

as gRPC resolver schemes share a common namespace, we should play nice and choose prefixed names